packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -203,52 +237,182 @@ class AdaptiveScaffold extends StatefulWidget {
   /// list of [NavigationDestination]s. Takes in a [selectedIndex] property for
   /// the current selected item in the [NavigationRail] and [extended] for
   /// whether the [NavigationRail] is extended or not.
-  static Builder toNavigationRail({
+  static Builder toRailFromDestinations({
     required List<NavigationDestination> destinations,
     double width = 72,
     int selectedIndex = 0,
     bool extended = false,
     Color backgroundColor = Colors.transparent,
+    EdgeInsetsGeometry padding = const EdgeInsets.all(8.0),
     Widget? leading,
     Widget? trailing,
     Function(int)? onDestinationSelected,
+    IconThemeData selectedIconTheme = const IconThemeData(color: Colors.black),
+    IconThemeData unselectedIconTheme =
+        const IconThemeData(color: Colors.black),
+    TextStyle selectedLabelTextStyle = const TextStyle(color: Colors.black),
+    NavigationRailLabelType labelType = NavigationRailLabelType.none,
+  }) {
+    return toNavigationRail(
+        width: width,
+        labelType: labelType,
+        leading: leading,
+        trailing: trailing,
+        onDestinationSelected: onDestinationSelected,
+        backgroundColor: backgroundColor,
+        padding: padding,
+        extended: extended,
+        selectedIndex: selectedIndex,
+        selectedIconTheme: selectedIconTheme,
+        unselectedIconTheme: unselectedIconTheme,
+        selectedLabelTextStyle: selectedLabelTextStyle,
+        destinations: destinations
+            .map((NavigationDestination e) => _toRailDestination(e))
+            .toList());
+  }
+
+  /// Public helper method to be used for creating a [NavigationRail] from a
```

Comment:
The first paragraph of any dartdoc section must be a short self-contained sentence that explains the purpose and meaning of the item being documented.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#dartdoc-specific-requirements